sql/core/src/test/scala/org/apache/spark/sql/streaming/FileStreamSourceSuite.scala

```diff
@@ -165,19 +177,32 @@ class FileStreamSourceSuite extends FileStreamSourceTest with SharedSQLContext {
       .collect { case s @ StreamingRelation(dataSource, _, _) => s.schema }.head
   }
 
-  test("FileStreamSource schema: no path") {
+  // ============= Basic parameter exists tests ================
+
+  test("FileStreamSource schema: no path, no schema") {
     val e = intercept[IllegalArgumentException] {
       createFileStreamSourceAndGetSchema(format = None, path = None, schema = None)
     }
-    assert("'path' is not specified" === e.getMessage)
+    assert(e.getMessage.contains("path"))  // reason is path, not schema
   }
 
-  test("FileStreamSource schema: path doesn't exist") {
-    intercept[AnalysisException] {
+  test("FileStreamSource schema: path doesn't exist, no schema") {
+    val e = intercept[IllegalArgumentException] {
       createFileStreamSourceAndGetSchema(format = None, path = Some("/a/b/c"), schema = None)
     }
```

Comment:
I think you should be able to create a file stream even when there is not yet any data present.